torch/csrc/jit/export.cpp

```diff
@@ -788,6 +796,9 @@ void ScriptModuleSerializer::convertModule(
     record->set_key(filename.str());
   }
 
+  // Run the '__getstate__' method on the module and store the result
```

Comment:
The behavior here seems weirdly organized. getstate should error if it doesn't exist and we should be checking here if it exists. Furthermore, if module has a getstate/setstate then we _should not_ serialize its attributes and tensors, I do not see code for that. It is likely that having getstate corresponds with having tensors and attributes that _are not serializable_ so this distinction is important.

Reply:
If we only serialize one or the other, how to we handle references to parameters/attributes? Should we be serializing only the proto stubs with the names/types and then leaving their values uninitialized and assume that `__setstate__` will handle them? Providing a `__getstate__` may also only mean that some of the attributes/tensors are not serializable, but maybe some of the others are. It seems potentially confusing if some features are being turned off by adding a `__getstate__` method.